File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int tar) {
        unordered_map<int,int> ma;
        for(int i=0;i<nums.size();i++){
            int rem = tar-nums[i];
            if(ma.find(rem)!=ma.end()){
                return vector<int>({ma[rem],i});
            }
            ma.insert({nums[i],i});
        }
        return vector<int>({});
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];
        HashMap<Integer,Integer> hm = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int complement = target - nums[i];
            if(hm.containsKey(complement)){
                ans[0]= hm.get(complement);
                ans[1]= i;
                break;
            }
            else{
                hm.put(nums[i], i);
            }
        }
        return ans;
    }
}